src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
 export class Triangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a < 1 || b < 1 || c < 1) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+
+    this.shape = 'triangle';
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const p = (a + b + c) / 2;
+    const area
+      = Math.floor(Math.sqrt(p * (p - a) * (p - b) * (p - c)) * 100) / 100;
 
+    return area;
+  }
 }
 
 export class Circle implements Figure {
+  shape: Shape;
 
+  constructor(public color: Color, public radius: number) {
+    if (radius < 1) {
+      throw new Error('radius is null');
+    }
+
+    this.shape = 'circle';
+  }
+
+  getArea(): number {
+    return Math.floor((Math.PI * this.radius ** 2) * 100) / 100;
+  }
 }
 
 export class Rectangle implements Figure {
+  shape: Shape;
+
+  constructor(
+    public color: Color,
+    public weidth: number,
+    public height: number,
+  ) {
+    if (weidth < 1 || height < 1) {
+      throw new Error('must be a size');
```

Comment:
this message is not very clear... it would be better to write something like this:
'All side lengths must be more than 0'

Check all your messages, maybe some texts can be improved